code/game/machinery/wishgranter.dm

```diff
@@ -39,4 +39,44 @@
 
 		user.mind.add_antag_datum(/datum/antagonist/wishgranter)
 
-		to_chat(user, "You have a very bad feeling about this.")
\ No newline at end of file
+		to_chat(user, "You have a very bad feeling about this.")
+		
+/obj/machinery/wish_granter/super
+	name = "super wish granter"
+	var/list/types = list()
+	
+/obj/machinery/wish_granter/super/attack_hand(mob/living/carbon/user)
+	if(!ishuman(user))
+		to_chat(user, "You feel a dark stirring inside of the Wish Granter, something you want nothing of. Your instincts are better than any man's.")
+		return
+	else if(is_special_character(user) || jobban_isbanned(user, ROLE_TRAITOR) || jobban_isbanned(user, "Syndicate"))
+		to_chat(user, "Something instinctual makes you pull away.")
+		return
+	else
+	
+		to_chat(user, "Your touch makes the Wish Granter stir.  Are you really sure you want to do this?")
+		
+		for(var/supname in GLOB.all_superheroes)
```

Comment:
for whatever reason
`for(var/datum/superheroes/supname in GLOB.all_superheroes)`
`    if(!supname.activated)`
`         types+= supname`
`didn't work whatsoever, even with the if statement removed